### PR TITLE
AL/BE-182. Create liquidation pools.

### PR DESCRIPTION
**Pull request by lut1k**
Added liquidation-pools pallet.
Added new trait PoolsManager for liquidity-pools and liquidation-pools.
Implmented mock file and tests file for liquidation-pools pallet.
Fixed tests after adding new trait PoolsManager.

### CODE REVIEW

File: pallets/liquidation-pools/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+use serde::{Deserialize, Serialize};
+use sp_runtime::traits::AccountIdConversion;
+use sp_runtime::{ModuleId, RuntimeDebug};
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+/// Liquidtaion Pool metadata
```
Comment (Apollo50) [inline pallets/liquidation-pools/src/lib.rs:25]: Liquidation*

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -248,6 +249,7 @@ impl m_tokens::Trait for Runtime {
 impl minterest_protocol::Trait for Runtime {
 	type Event = Event;
 	type Borrowing = liquidity_pools::Module<Runtime>;
+	type ManagerLiquidityPools = LiquidityPools;
```
Comment (Apollo50) [inline runtime/src/lib.rs:252]: maybe better to name types for this trait equally for protocol & controller. (protocol -> ManagerLiquidityPools, controller -> LiquidityPoolsManager )
and use alias for borrowing trait as for manager trait

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -293,6 +295,7 @@ impl liquidity_pools::Trait for Runtime {
 
 impl controller::Trait for Runtime {
 	type Event = Event;
+	type LiquidityPoolsManager = LiquidityPools;
```
Comment (Apollo50) [inline runtime/src/lib.rs:298]: ^^

**Review by Apollo50 — APPROVED** [review]

File: pallets/liquidation-pools/src/lib.rs (hunk 1 above)
Comment (lut1k) [reply pallets/liquidation-pools/src/lib.rs:25]: Fixed

File: runtime/src/lib.rs (hunk 2 above)
Comment (lut1k) [reply runtime/src/lib.rs:252]: Fixed alias for borrowing trait as for manager trait.
I cannot use the same names for generic types. Pallet minterest-protocol uses the controller pallet. I am getting an ambiguous link.

File: runtime/src/lib.rs (hunk 3 above)
Comment (lut1k) [reply runtime/src/lib.rs:298]: I cannot use the same names for generic types. Pallet minterest-protocol uses the controller pallet. I am getting an ambiguous link.

**Review by Pavel-Blaize — APPROVED** [review]

**Pull request merged**